modules/market/src/main/java/com/opengamma/strata/market/curve/meta/TenorNoDateCurveNodeMetadata.java

```diff
@@ -0,0 +1,374 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.curve.meta;
+
+import java.io.Serializable;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutablePreBuild;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.date.Tenor;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.market.curve.CurveParameterMetadata;
+
+/**
+ * Curve node metadata for a curve node with a specific tenor and no date.
+ */
+@BeanDefinition(builderScope = "private")
+public final class TenorNoDateCurveNodeMetadata
```

Comment:
Could we call this `TenorCurveNodeMetadata` and rename the existing one `TenorDateCurveNodeMetadata`?